backend/inspirehep/matcher/api.py

```diff
@@ -23,11 +25,19 @@ def get_reference_from_grobid(query):
     return GrobidReferenceParser(response.text).parse()
 
 
+def _update_search_index_in_configs(configs):
```

Comment:
I would avoid `deepcopy` etc. We could pull out the logic of https://github.com/inspirehep/inspirehep/blob/08ae47665713aa0cce90572b72e7f2eb710cd0d6/backend/inspirehep/indexer/cli.py#L325 and create a utility function something like:

```python
def get_prefixed_index(index_name, index_prefix=None):
    prefix = index_prefix or current_app.config.get("SEARCH_INDEX_PREFIX")
    if prefix not in index_name:
          return prefix + index_name
    return index_name
```

then we can reuse it in the `mathcer/config.py` for example:

```python
CONFIG = {    
  "algorithm": [
        {
            "queries": []
           
    ],
    "index": get_prefixed_index("records-hep"),
}
```